docs/proposals/maintenance.md

```diff
@@ -0,0 +1,216 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/maintenance.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Node Maintenance
+
+## Abstract
+
+A proposal for implementing safe node maintenance primitives, in stages.
+
+## Overview
+
+Nodes need periodic maintenance, most of which require partial or total
+unavailability to pods.  The cluster admin would like for this to happen (1)
+automatically, and (2) respecting any service quality constraints.
+
+Examples of maintenance:
+
+- unmount/remount filesystems to change mkfs params
+- remove from service but leave up for debugging
+- reboot for kernel or image upgrade
+- remove from service for hardware repair
+- gracefully reducing the size of the cluster
+- take down machines all machines in a rack for maintenance
+
+## Service Quality
+
+This may be thought of as the cluster admin offering a guarantee or SLO, or as the
+user specifying their requirements.  Around evictions, there are typically three
+parameters:
+
+- time between evictions (of pods in the same replica set)
```

Comment:
It would be good to explain the reason for each of these service quality parameters.
e.g.: ensuring time between evictions allows a replacement pod some time to warm up and rebuild its caches before proceeding to the next victim; eviction notice allows a pod to write out a checkpoint, finish outstanding requests and background processing, etc;